### ISSUE TITLE
Functional test suite

### ISSUE THREAD

**Issue by jgonggrijp** [NBOCampbellToets/CampbellSoup#12]
Probably using PhantomJS and possibly Protractor.

**Comment by jgonggrijp** [comment]
After some study, I'm homing in on [`grunt-protractor-runner`](https://www.npmjs.com/package/grunt-protractor-runner), initially using [this page](http://ng-learn.org/2014/02/Protractor_Testing_With_Angular_And_Non_Angular_Sites/) as a source of hints.

If that turns out unsuccessful, I will probably go with [Casper](http://docs.casperjs.org/en/latest/testing.html). It is actively maintained and works with PhantomJS (which we already use).

There are some Grunt plugins for Casper available, but unfortunately, none of them appears to be well maintained or widely used. Among them, [`grunt-casperjs`](https://www.npmjs.com/package/grunt-casperjs) seems to be the most promising; we could try that first. If it does not work, it should be possible to create our own wrapper using `grunt-shell`.

**Comment by jgonggrijp** [comment]
839cbf0

I decided to go with CasperJS first, anyway. A Protractor-based solution could potentially be run against more different stock browsers; CasperJS only supports headless WebKit and headless Gecko (SlimerJS), where the latter is supported only experimentally. However, CasperJS requires fewer software layers, which means there are fewer potential sources of headaches as well. End-to-end testing against real-world browsers is definitely necessary, but I think it can be left to human testers at a later stage, when all headless functional tests have already succeeded.

In the commit above I did run into one issue. If this turns out to be hard to solve, I will try my luck with Protractor after all.
